application.go

```diff
@@ -104,6 +77,92 @@ func IsApplicationEqual(a, b Application) bool {
 		a.Handlers().IsEqual(b.Handlers())
 }
 
+// ForeignMessageNames returns the subset of message names used by an
+// application that must be communicated beyond the scope of that application.
+//
+// This includes:
+//  - commands that are produced by the application, but consumed elsewhere
+//  - commands that are consumed by the application, but produced elsewhere
+//  - events that are consumed by the application, but produced elsewhere
+func ForeignMessageNames(app Application) EntityMessageNames {
+	m := app.MessageNames()
+	f := EntityMessageNames{
+		Roles:    message.NameRoles{},
+		Produced: message.NameRoles{},
+		Consumed: message.NameRoles{},
+	}
+
+	for mt, r := range m.Produced {
```

Comment:
This is a minor thing, but is `mt` a good name for that var. These are "message names" right?
I see the `mt` "message type" version in the other function below.

Reply:
Yep, fair call, this is a typical copy/paste error >_<

Reply:
I've fixed these, improved the comments, and collapsed some verbose logic down a little. Please re-review.